### ISSUE TITLE
srsRAN gNB with srsUE over usrp b210 frontends

### ISSUE THREAD

**Issue by thaxir** [srsran/srsRAN_Project#933]
## Issue Description ##
[Describe the issue in detail]
I'm following the recent tutorial 'srsRAN gNB with srsUE' from here https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html. However, the UE (srsue) never obtains an IP from srsgnb staying at this :

```RRC Connected
RF status: O=3, U=0, L=0
Scheduling request failed: releasing RRC connection...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=1934
```

## Setup Details ##
[Specify details of the test setup. This would help us reproduce the problem reliably]
e.g. Network configuration, Operation System, Hardware, RF front-end, library and driver versions

2 USRP b210 as RF frontend
2 NUCs
Open5gs
srsRAN_Project for gnb
srsRAN_4G for the UE
Internal clock/no GPSDO

## Expected Behavior ##
[What you expect to happen]
Cannot use srsue since it is not properly connected to the gnb
## Actual Behaviour ##
[What happens instead e.g. error message]
Seems like  UE attempted a scheduling request but didn't receive a response from the gNB. Not sure though
![image](https://github.com/user-attachments/assets/7a8c8253-241a-427f-a618-191009281ac9)

## Steps to reproduce the problem ##
[Tell us how to reproduce this issue e.g. RF setup, application config files]

## Additional Information ##
[Any additional information, configuration or data that might be necessary to reproduce the issue]
[gnb.log.txt](https://github.com/user-attachments/files/17734391/gnb.log.txt)
[ue.log.txt](https://github.com/user-attachments/files/17734395/ue.log.txt)

[ue.conf.txt](https://github.com/user-attachments/files/17734375/ue.conf.txt)

[gnb-config.yml.txt](https://github.com/user-attachments/files/17734371/gnb-config.yml.txt)

**Comment by pgawlowicz** [comment]
UL signal is too strong at the gnb. Please try to tune down srsUE tx_gain and gnb rx_gain.

**Comment by thaxir** [comment]
any suggestions on this topic ??

**Comment by pgawlowicz** [comment]
could you try to apply the patch from this comment to srsUE, recompile and try again? 
https://github.com/srsran/srsRAN_Project/issues/826#issuecomment-2370588502

**Comment by thaxir** [comment]
@pgawlowicz  Piotr  we have done the changes you suggested .
 After restarting serveral times  srsue and gnb the srsue   logs are changed as you see in the snapshot below
![image](https://github.com/user-attachments/assets/3c21ae28-d523-46ac-8f27-a5e28ebf9f91)

Gnb reports this:
![image](https://github.com/user-attachments/assets/ee7d02c8-41b8-4b86-8611-0c6602744e5d)

and amf in open5gs which is in the same machine with gnb reports this :
![image](https://github.com/user-attachments/assets/c977d33e-1da7-4e7c-99e6-95f16f140bf0)

which seems that amf rejects the srsue . Do you know why? (

And another question is why do i need to restart several times both srsue and gnb in order to have this connection open and not the **Scheduling request failed : releasing RRC connection**?
I attach also the logs  for ue and gnb.
[ue.log.3.txt](https://github.com/user-attachments/files/17845645/ue.log.3.txt)
[gnb.log.3.txt](https://github.com/user-attachments/files/17845726/gnb.log.3.txt)

[gnb.log.3.txt](https://github.com/user-attachments/files/17845691/gnb.log.3.txt)

**Comment by thaxir** [comment]
Any suggestions  on this?

**Comment by pgawlowicz** [comment]
it seems the problem is on the core network side, did you add the UE to the subscriber database?

**Comment by thaxir** [comment]
Hi @pgawlowicz  again . So far i have tried to experiment with  the position of the usrps  if the radio link and the line of site was the error but i am getting on the srsue
![image](https://github.com/user-attachments/assets/4578441d-2cf1-4819-9b2a-a01628ecb83c)

On the gnb 
![image](https://github.com/user-attachments/assets/82894342-e263-4b31-b99e-27bcfa740a96)

I was able to get some movement over the amf

![image](https://github.com/user-attachments/assets/4e779369-830d-498e-be0a-69c21efe9eee)



 i started cross checking the database i see the same  values in the srsue and open5gs (deleting and re-adding the subscriber)
 
![image](https://github.com/user-attachments/assets/95cdff2f-c06b-426c-ad72-14bf89a2b272) 
![image](https://github.com/user-attachments/assets/abedd2e2-7f45-4886-a2f7-5c2da8775482)


Searching for others with this issue i came across this https://github.com/srsran/srsRAN_Project/issues/451 which seems to have a lot of similarities with my case.  However, on the newer version of srsran gnb  i don't  know where to put the ``` inactivity_timer: 7200```

Does this inactivity timer can be added somewhere in the gnb ?

**Comment by pgawlowicz** [comment]
could you share gnb logs in debug mode?

**Comment by thaxir** [comment]
@pgawlowicz  yes here 
i don t  know if i have captured all because there are a lot of msgs.
[gnb-debug.txt](https://github.com/user-attachments/files/17977674/gnb-debug.txt)

**Comment by thaxir** [comment]
Apologies, 
here
[gnb-debug.txt](https://github.com/user-attachments/files/17977854/gnb-debug.txt)

**Comment by pgawlowicz** [comment]
Could you tell me which code version (commit id) do you use? it is missing in the logs.

**Comment by thaxir** [comment]
How to i see this?

**Comment by pgawlowicz** [comment]
git log -n1

**Comment by thaxir** [comment]
![image](https://github.com/user-attachments/assets/81dfb582-1924-40b6-a952-82abe3b9ffd6)
Merge: eea87b1d8 f8978a697

**Comment by thaxir** [comment]
Should i use  binaries from  Package Manager on ubuntu or build it from the source?

**Comment by pgawlowicz** [comment]
both ways  work

**Comment by thaxir** [comment]
@pgawlowicz i just installed  srsran-project  (apt-get install)  from the package manager. 
nuc6@nuc6:/usr/share/srsran$ gnb --version
srsRAN 5G gNB version 23.10.1 ()

Should i proceed to testing ?

**Comment by thaxir** [comment]
Ok done 
![image](https://github.com/user-attachments/assets/e8f18d5b-7d56-4f2a-b034-36c07c69a6e7)


No i get  this
![image](https://github.com/user-attachments/assets/f0e34c8e-25a9-49e7-a342-98628fdc835b)

It seems like a config problem 
Here is the config of gnb 
![image](https://github.com/user-attachments/assets/a06cf7b9-cd67-4767-8540-9269d3179bbf)

**Comment by pgawlowicz** [comment]
You have to install uhd driver if you are building from source. Please follow our installation guide https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html

**Comment by thaxir** [comment]
I have already installed the uhd from the previous run 
![image](https://github.com/user-attachments/assets/4aa0ec30-e689-4d82-a0af-d3d6f273771a)

Do i need to redo this installation?

**Comment by pgawlowicz** [comment]
could you check if cmake detects the UHD correctly?

**Comment by thaxir** [comment]
No it is not detected so i build 3.15 again . I will send you the update tomorrow. Thank you very much this far for the help

**Comment by thaxir** [comment]
Hi @pgawlowicz again, so the update . I have installed uhd 3.15 .
Gnb : 
![image](https://github.com/user-attachments/assets/739d1895-ef7d-43c2-90b0-361ff19ad79e)
Here the logs 
[gnb-debug(2).txt](https://github.com/user-attachments/files/17992430/gnb-debug.2.txt)


The srsue 
![image](https://github.com/user-attachments/assets/8dd67ffc-dee4-4645-826b-61ca5c4a73dd)

**Comment by pgawlowicz** [comment]
and the srsUE config and logs?

**Comment by thaxir** [comment]
Ue logs:
[ue.txt.zip](https://github.com/user-attachments/files/17992751/ue.txt.zip)
Ue conf:
[ue-conf.txt](https://github.com/user-attachments/files/17992777/ue-conf.txt)

**Comment by pgawlowicz** [comment]
PUSCH SNR is very low, probably caused by a very high CFO. Do you have an external clock to connect to b210s?

**Comment by thaxir** [comment]
No i don t have ext clock. I could order  one   but it will take time . I don t know if i could  connect the two usrps through  coaxial cable for using on usrp as master timer and one as slave . 
Is there any configuration i can do now without a gpsdo?

**Comment by pgawlowicz** [comment]
you might try to tune TX an RX gains for both gnb and srsUE, also time_adv_nsamples in srsUE config. Some users are able to run this setup without an external clocks. Did you check if it works correctly over zmq?

**Comment by thaxir** [comment]
Yes in the past i did use zmq before shifting to ueransim. For the time being i need to have radio link between ue and gnb. Do you have any reference values for start playing with the gnb /ue /time_adv ?
If i replace srsue with a cots ue would this still be a problem?

**Comment by pgawlowicz** [comment]
Usually, `tx_gain: 60` and `rx_gain:30-40` and 'time_adv = auto or 300'. Did you apply the srsUE patch that I shared in one of the previous posts?
I think b210 should work correctly with COTS. You might give it a try.
